storage/storage-samples/storage-di-sample/src/main/java/org/fido/iot/storage/DiDbStorage.java

```diff
@@ -77,36 +75,68 @@ public Composite createVoucher(Object createParams) {
       throw new InvalidMessageException("mino must be an array");
     }
 
+    // Determine type of device and create appropriate voucher
+    // The M string (mstr) is a CBOR array of several values:
+    // FIRST_KEY (number): the key type requirement of the device, as an integer-coded KeyType
+    //   This is the type of owner key the device is prepared to parse, not the type
+    //   of the device's key.
+    // SECOND_KEY (string): the device serial number
+    // THIRD_KEY (string): a device info hint
+    // FOURTH_KEY (bstr):
+    // - if the device is using an EC keypair, a CSR.
+    // - if the device is OnDie ECDSA then the device cert chain and a
+    //   test signature is also present
+    // FIFTH_KEY (bstr): present only for OnDie ECDSA: the test signature
+
     Composite mstr = Composite.fromObject(createParams);
 
     voucher = Composite.newArray();
     Composite header = Composite.newArray();
     String customerId = null;
 
-    //setup cert hash and public key
-    Composite chain = null;
-    Composite chainHash = null;
-
     Composite settings = getSettings();
     Integer validityDays = settings.getAsNumber(Const.FIRST_KEY).intValue();
     if (settings.get(Const.THIRD_KEY) != null
         && settings.get(Const.THIRD_KEY) != Optional.empty()) {
       customerId = settings.getAsString(Const.THIRD_KEY);
     }
 
+    //setup cert hash and public key where applicable
+    Object publicKey = null;
+    Composite chainHash = null;
+    Composite chain = null;
+
     UUID guid = UUID.randomUUID();
-    Certificate[] issuerChain = resolver.getCertChain(
-        mstr.getAsNumber(Const.FIRST_KEY).intValue());
 
+    int keyType = mstr.getAsNumber(Const.FIRST_KEY).intValue();
     String serialNo = mstr.getAsString(Const.SECOND_KEY);
 
-    if (mstr.get(Const.FOURTH_KEY) != Optional.empty()) {
+    if (keyType == Const.PK_SECP256R1 || keyType == Const.PK_SECP256R1) {
```

Comment:
Shouldn't one of the two constants be `PK_SECP384R1`?

Reply:
Good catch, one should be 384.